fluent-syntax/src/ast.ts

```diff
@@ -5,10 +5,11 @@
  * Annotation.
  *
  */
-export class BaseNode {
-  constructor() {}
+export abstract class BaseNode {
+  public type = "BaseNode";
+  [name: string]: unknown;
```

Comment:
It's an [index signature](https://www.typescriptlang.org/docs/handbook/interfaces.html#indexable-types). The `Visitor` needs it because it dynamically accesses methods based on the type of the visited node. I've used `unknown` so that the subclasses of `BaseNode` can narrow down the types of their members according to their needs.